spec/helpers/application_helper_spec.rb

```diff
@@ -336,30 +356,44 @@
       context "when work becomes anonymous" do
         let(:collection) { create(:anonymous_collection) }
 
+        before do
+          work.users.each { |user| user.preference.update(allow_collection_invitation: true) }
+        end
+
         it "returns updated string" do
-          travel_to(1.day.ago)
-          original_cache_key = "#{work.cache_key_with_version}/blurb_css_classes-v2"
-          expect(helper.css_classes_for_creation_blurb(work)).to eq("#{default_classes} work-#{work.id} user-#{user1.id}")
+          original_cache_key = nil
+          travel_to(1.day.ago) do
+            original_cache_key = "#{work.cache_key_with_version}/blurb_css_classes-v2"
+            expect(helper.css_classes_for_creation_blurb(work)).to eq("#{default_classes} work-#{work.id} user-#{user1.id}")
+          end
 
-          travel_back
-          work.collections << collection
-          expect(helper.css_classes_for_creation_blurb(work)).to eq("#{default_classes} work-#{work.id}")
-          expect(original_cache_key).not_to eq("#{work.cache_key_with_version}/blurb_css_classes-v2")
+          travel_to(1.second.from_now) do
```

Comment:
I'm not really sure why this became necessary to get the test passing; I suspect it has something to do with how `in_unrevealed_collection?` gets updated, but I couldn't find exactly where it happens. Just wanted to flag this in case it's a sign of a bigger issue.

Reply:
Variables defined with `let` are lazily evaluated. In the original version of this test, the first time `work` was referenced was in the `travel_to(1.day.ago)` block, meaning that the work got created (and most recently updated) 1 day ago. That meant that when the work was updated again in the present, its update date changed, and the cache on its CSS classes expired.

By adding a `before` hook that references `work`, you're causing the work to be created before the `travel_to(1.day.ago)` hook, making the work be created in the present. This makes the first `travel_to` block a no-op, and means that if the test runs fast enough you're just creating the work and then updating it all in the same second, and nothing gets expired.

(Luckily, you should be able to revert almost all of the spec changes once you pull the validation out of the model.)

Reply:
Thank you so much for the explanation!